src/client/networkclient.cc

```diff
@@ -0,0 +1,371 @@
+  return client_description_;
+}
+
+QString NetworkClient::clientType() {
+  return client_type_;
+}
+
+QString NetworkClient::authenticationKey() {
+  return authentication_key_;
+}
+
+QTextStream* NetworkClient::output() {
+  return output_stream_;
+}
+
+void NetworkClient::setOutput(QTextStream* stream) {
+  output_stream_ = stream;
+}
+
+void NetworkClient::sendMsgConnect() {
+  std::shared_ptr<std::string> client_name_ptr(
+      new std::string(client_name_.toStdString()));
+  std::shared_ptr<std::string> client_version_ptr(
+      new std::string(client_version_.toStdString()));
+  std::shared_ptr<std::string> client_description_ptr(
+      new std::string(client_description_.toStdString()));
+  std::shared_ptr<std::string> client_type_ptr(
+      new std::string(client_type_.toStdString()));
+  std::shared_ptr<std::string> authentication_key_ptr(
+      new std::string);
+
+  msg_ptr msg(new messages::MsgConnect(
+      1,
+      pair_str(true, client_name_ptr),
+      pair_str(true, client_version_ptr),
+      pair_str(true, client_description_ptr),
+      pair_str(true, client_type_ptr),
+      pair_str(true, authentication_key_ptr)
+      ));
+
+  sendMessage(msg);
+}
+
+void NetworkClient::sendMessage(msg_ptr msg) {
+  if (client_socket_ && client_socket_->isValid()) {
+    msgpack::sbuffer buf;
+    msgpack::packer<msgpack::sbuffer> packer(buf);
+    messages::MsgpackWrapper::dumpObject(packer, msg);
+    client_socket_->write(buf.data(), buf.size());
+  }
+}
+
+void NetworkClient::registerMessageHandlers() {
+  message_handlers_["sub_cancelled"] = &NetworkClient::handleTreeModificationMessage;
+  message_handlers_["plugin_trigger_done"] = &NetworkClient::handlePluginTriggerDoneMessage;
+  message_handlers_["get_reply"] = &NetworkClient::handleTreeModificationMessage;
+  message_handlers_["connected"] = &NetworkClient::handleConnectedMessage;
+  message_handlers_["proto_error"] = &NetworkClient::handleProtoErrorMessage;
+  message_handlers_["connections_reply"] = &NetworkClient::handleConnectionsMessage;
+  message_handlers_["registry_reply"] = &NetworkClient::handleRegistryReplyMessage;
+  message_handlers_["mthd_res"] = &NetworkClient::handleMthdResMessage;
+  message_handlers_["plugin_trigger_run"] = &NetworkClient::handlePluginTriggerRunMessage;
+  message_handlers_["modify_err"] = &NetworkClient::handleTreeModificationMessage;
+  message_handlers_["list_reply"] = &NetworkClient::handleTreeModificationMessage;
+  message_handlers_["get_data_reply"] = &NetworkClient::handleTreeModificationMessage;
+  message_handlers_["ack"] = &NetworkClient::handleTreeModificationMessage;
+  message_handlers_["get_bindata_reply"] = &NetworkClient::handleTreeModificationMessage;
+  message_handlers_["obj_gone"] = &NetworkClient::handleTreeModificationMessage;
+  message_handlers_["create"] = &NetworkClient::handleTreeModificationMessage;
+  message_handlers_["conn_error"] = &NetworkClient::handleConnErrorMessage;
```

Comment:
Wrong message type here - should be `connection_error`.
Overall it looks like more of those are wrong as well.